rflx/model/message.py

```diff
@@ -2222,6 +2277,7 @@ def _merge_inner_message(
                         merged_condition = expr.And(
                             link.condition, final_link.condition
                         ).substituted(mapping=substitution)
+                        merged_condition = merged_condition.substituted(set_types)
```

Comment:
Why do the types need to be set here? Shouldn't `link.condition` and `final_link.condition` be already correctly typed?

Reply:
The inner message already contains the merged field names, e.g. `F1_F3`. However before merging it doesn't contain the correct types for the fields so the types have to be recalculated after knowing the type mappings of both the outer and the inner message.

Reply:
Why don't they contain the correct types? The field types shouldn't change by the merging. Isn't the actual problem here that `prefixed` doesn't preserve the types of the variables?

Reply:
I added a clarification.